### ISSUE TITLE
No GitHub token found.

### ISSUE THREAD

**Issue by mvds314** [CopilotC-Nvim/CopilotChat.nvim#409]
I am using this plugin under Windows 11 (Neovim 0.10.1) and the copilot.vim plugin.

I keep gettting the following error:
> No GitHub token found, please use `:Copilot auth` to set it up from copilot.lua or `:Copilot setup` for copilot.vim

Copilot itself works.
On my Linux machine, the plugin _does work_ with this config.

I use the following config for the plug:
```
{
    "CopilotC-Nvim/CopilotChat.nvim",
    branch = "canary",
    dependencies = {
      -- { "zbirenbaum/copilot.lua" }, -- or github/copilot.vim
      { "github/copilot.vim" },
      { "nvim-lua/plenary.nvim" }, -- for curl, log wrapper
    },
    -- build = "make tiktoken", -- Only on MacOS or Linux
    -- build = vim.fn.has("win32") == 1 and "powershell -ExecutionPolicy Bypass -File Build-LuaTiktoken.ps1" or "make"
    -- opts = {
    -- debug = true, -- Enable debugging
    -- See Configuration section for rest
    -- },
    config = function()
      require("CopilotChat").setup {
        debug = true, -- Enable debugging
        -- See Configuration section for rest
      }
    end,
    lazy = false,
    -- See Commands section for default commands if you want to lazy load on them
},
```
My complete config can be found [here ](https://github.com/mvds314/NvChadConfig/blob/myNvChadConfig/lua/plugins/init.lua).

I tried a couple of things:
- enable/disable the tiktoken
- Reinstall the plugin
- Reinstall copilot.vim

Anyone any ideas on how I can make this plugin work?

**Comment by acheong08** [comment]
Does `~/.config/github-copilot/hosts.json` exist?

**Comment by mvds314** [comment]
Yes, on my windows machine it's located under: `c:\Users\MyUsername\AppData\Local\github-copilot\hosts.json`

It has the following content:
> {"github.com":{"user":"<my_github_username>","oauth_token":"<token>"}}

Only my linux machine it looks the same, only the token is different.

**Comment by acheong08** [comment]
https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/d43fab67c328946fbf8e24fdcadfdb5410517e1f/lua/CopilotChat/copilot.lua#L78C16-L113

Could it be how the paths are separated? Your setup looks about right to me.

**Comment by mvds314** [comment]
I don't know, my programming knowledge within the neovim infrastructure is limited.
What I do know is that, under windows, `~` does not redirect to the home folder. Maybe it does if you use it in lua when writing neovim plugins.

To test it, can I call this function in Neovim like this?

> `:lua require("CopilotChat.copilot").find_config_path()`

**Comment by mvds314** [comment]
Okay, slowly getting there... I just manually changed the line 83  `config = vim.fn.expand('~/AppData/Local')`
Copilot thinks it should be this ;-)
`config = vim.fn.expand('$LOCALAPPDATA')`

Now I'm getting another error:

> Failed to get response: {
  exit = 35,
  message = 'get https://api.github.com/copilot_internal/v2/token - curl error exit_code=35 stderr={ "curl: (35) schannel: next InitializeSecurityContext failed: CRYPT_E_NO_REVOCATION_CHECK (0x80092012) - The revocation function was unable to check revocation for the certificate." }',
  stderr = '{ "curl: (35) schannel: next InitializeSecurityContext failed: CRYPT_E_NO_REVOCATION_CHECK (0x80092012) - The revocation function was unable to check revocation for the certificate." }'
}
```

**Comment by mvds314** [comment]
Solution to the curl problem can be found [here](https://github.com/CopilotC-Nvim/CopilotChat.nvim/issues/375).

Just set: `allow_insecure = true`

**Comment by deathbeam** [comment]
Alright, created PR for the fix (with fallback to original value when not found)
